android-components/plugins/config/src/main/java/ConfigPlugin.kt

```diff
@@ -52,6 +53,112 @@ object Config {
         return readVersionFromFile().split(".")[0]
     }
 
+    /**
+     * Generate a build date that follows the ISO-8601 format
+     */
+    @JvmStatic
+    fun generateBuildDate(): String {
+        return LocalDateTime.now().toString()
+    }
+
+    private val fennecBaseVersionCode by lazy {
+        val format = SimpleDateFormat("yyyyMMddHHmmss", Locale.US)
+        val cutoff = format.parse("20141228000000")
+        val build = Date()
+
+        Math.floor((build.time - cutoff.time) / (1000.0 * 60.0 * 60.0)).toInt()
+    }
+
+    /**
+     * Generates a versionCode that follows the same rules like legacy Fennec builds.
+     * Adapted from:
+     * https://searchfox.org/mozilla-central/rev/34cb8d0a2a324043bcfc2c56f37b31abe7fb23a8/python/mozbuild/mozbuild/android_version_code.py
+     *
+     * There is a discrepancy between the epoch date used here (20141228)
+     * and the epoch used in Fennec (20150801) for historical reasons. We keep
+     * this discrepancy to avoid having Fenix version codes decrease.
+     * Note that the original Fennec implementation also had an inconsistency in
+     * the documented epoch date (20150901) and the effective epoch date (20150801).
+     */
+    @JvmStatic
+    fun generateFennecVersionCode(abi: String): Int {
```

Comment:
@jonalmeida I am wondering if you think we really need this piece of code anymore? I see in Focus that we have a different approach https://searchfox.org/mozilla-mobile/source/firefox-android/focus-android/tools/gradle/versionCode.gradle#41

Reply:
Yes, we will need this until the end of time and should never remove this. We will not be able to upload to the play store if our version codes conflict with a previous version code that was added during fennec's lifetime.